cmd/block-generator/runner/run.go

```diff
@@ -127,11 +230,12 @@ func (r *Args) runTest(indexerURL string, generatorURL string) error {
 	start := time.Now()
 	for time.Since(start) < r.RunDuration {
 		time.Sleep(r.RunDuration / 10)
-		if err := collector.Collect("indexer_daemon_import_time_sec"); err != nil {
+
+		if err := collector.Collect(metrics.AllMetricNames...); err != nil {
 			return fmt.Errorf("problem collecting metrics: %w", err)
 		}
 	}
-	if err := collector.Collect("indexer_daemon_import_time_sec"); err != nil {
+	if err := collector.Collect(metrics.AllMetricNames...); err != nil {
```

Comment:
I'm searching for a substring so a map wouldn't work. The metric prefix should be configurable, we just haven't done it yet. Also, different metrics have different suffixes added automatically.

I don't think we need to do better than this. If we wanted to do better we'd probably replace the whole thing instead of optimizing this code.